main.tf

```diff
@@ -45,26 +22,59 @@ moved {
 }
 */
 
+locals {
+  # This is a big hack to enable us to generate something close to a custom error message
+  force_destroy_error_message = <<-EOT
+
+    ** ERROR: You must set `force_destroy_enabled = true` to enable `force_destroy`. **n/
+    ** WARNING: Upgrading this module from a version prior to 0.27.0 to this version **n/
+    **  will cause Terraform to delete your existing S3 bucket CAUSING COMPLETE DATA LOSS **n/
+    **  unless you follow the upgrade instructions on the Wiki [here](https://github.com/cloudposse/terraform-aws-s3-log-storage/wiki/Upgrading-to-v0.27.0-(POTENTIAL-DATA-LOSS)). **n/
+    **  See additional instructions for upgrading from v0.27.0 to v0.28.0 [here](https://github.com/cloudposse/terraform-aws-s3-log-storage/wiki/Upgrading-to-v0.28.0-and-AWS-provider-v4-(POTENTIAL-DATA-LOSS)). **n/
+
+    EOT
+  force_destroy_safety = {
+    true = {
+      true  = "true"
+      false = "false"
+    },
+    false = {
+      true  = local.force_destroy_error_message
+      false = "false"
+    }
+  }
+  # Generate an error message when `force_destroy == true && force_destroy_enabled == false`
+  force_destroy = tobool(local.force_destroy_safety[var.force_destroy_enabled][var.force_destroy])
+
+  bucket_name = var.bucket_name == null || var.bucket_name == "" ? module.this.id : var.bucket_name
+}
+
 module "aws_s3_bucket" {
   source  = "cloudposse/s3-bucket/aws"
-  version = "0.47.1"
+  version = "0.49.0"
 
-  bucket_name        = module.this.id
+  bucket_name        = local.bucket_name
   acl                = var.acl
-  force_destroy      = var.force_destroy
-  policy             = var.policy
+  force_destroy      = local.force_destroy
   versioning_enabled = var.versioning_enabled
 
-  lifecycle_rule_ids = [module.this.id]
-  lifecycle_rules    = [local.lifecycle_rule]
+  source_policy_documents = var.source_policy_documents
+  # Support deprecated `policy` input
```

Comment:
this deprecated input actually causes permanent drift with `terraform plan`.
On internet, there are many mentions that the input must not be used at all.

Reply:
@aknysh These are inputs to our `s3-bucket` module, not the AWS resource. Not a cause of drift.

Reply:
not saying we need to remove this in this PR, but we need to review the bucket module